### PR TITLE
Force user to change initial password.

### PR DESCRIPTION
**Pull request by andorp**
After login it redirect to the user settings page form the spaces and the delegations pages.

### CODE REVIEW

File: src/Frontend.hs [hunk 1]
```diff
@@ -172,15 +172,15 @@ type AulaMain =
 
 aulaMain :: ActionM m => ServerT AulaMain m
 aulaMain =
-       runHandler Page.viewRooms
+       forceChangeInitialPassword (runHandler Page.viewRooms)
```
Comment (fisx) [inline src/Frontend.hs]: why these two?  i'm particularly confused by the one wrapping the delegation network.  is that needed?

shouldn't we redirect to a dedicated end-point after login, and that end-point redirects to the actual landing page, whichever that is?  that way if we re-visit that page the check won't happen again, only after the next login.

File: src/Frontend/Core.hs [hunk 2]
```diff
@@ -431,6 +432,15 @@ runHandler mp = do
     handleDenied s u = throwServantErr $ (maybe Servant.err403 err303With u) { errBody = cs s }
         -- FIXME log these events as INFO, should we do this here or more globally for servant errors.
 
+-- ASSUMPTION: The user is logged in.
+forceChangeInitialPassword :: ActionM m => m a -> m a
+forceChangeInitialPassword handler = do
+    x <- handler
+    user <- currentUser
+    if has (userSettings . userSettingsPassword . _UserPassInitial) user
+        then do addMessage "Please change your password." -- TODO: Translation
```
Comment (fisx) [inline src/Frontend/Core.hs]: "Bitte ändere dein Passwort, damit niemand dein Login klauen kann."

**Comment by fisx** [comment]
this is related to #556, but it does not offer the delegation page yet, so we should leave that open.

i had a few comments, otherwise LGTM.  @andorp if you have time before monday you can fix it, otherwise first come first serve i guess.  thanks!

File: src/Frontend.hs (hunk 1 above)
Comment (andorp) [inline src/Frontend.hs]: My first idea was that. I wanted to code that into the `runHandler`. What would happen with that approach:

 1. Render login page
 1. User logs in
 1. runHandler on list ideas checks if there is an initial password, it redirects to the user setting page
 1. the server receives a GET request for user settings page
 1. runHandler on the user profile page checks if there is initial password, it redirects to the user settings page
 1. and so on

Comment (andorp) [inline src/Frontend.hs]: We could save as a session state if we rendered redirected once in the session on case of initial password.

Comment (andorp) [inline src/Frontend.hs]: Guarded list idea spaces and not guarded delegation network pages made me feel wierd as I tested out manually:

 - I can't open idea lists page but I can see my delegation? Hmm that's wierd.

I aimed the consistency and force the user to make change his initial password.

Comment (fisx) [inline src/Frontend.hs]: note that "profile page" and "settings page" are two different things.

but i still don't get it.  i guess i'll try what i have in mind and i'll see what you mean in 10 minutes (-:

**Comment by fisx** [comment]
@andorp what do you think?

**Comment by andorp** [comment]
I like it.

**Comment by andorp** [comment]
LGTM

**Pull request merged**